SU2_CFD/src/fluid/CNEMOGas.cpp

```diff
@@ -138,4 +141,150 @@ su2double CNEMOGas::GetrhoCvve() {
+  Ref_Temperature    = GetRefTemperature();
+  
+  /*--- Rename for convenience ---*/
+  rhoCvtr = V[RHOCVTR_INDEX];
+  rhoCvve = V[RHOCVVE_INDEX];
+
+  /*--- Pre-compute useful quantities ---*/
+  CvtrBAR = 0.0;
+  sqvel   = 0.0;
+  conc    = 0.0;
+  for (iDim = 0; iDim < nDim; iDim++)
+    sqvel += V[VEL_INDEX+iDim] * V[VEL_INDEX+iDim];
+  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+    CvtrBAR += rhos[iSpecies]*Cvtrs[iSpecies];
+    conc    += rhos[iSpecies]/MolarMass[iSpecies];
+  }
+
+  /*--- Species density derivatives ---*/
+  for (iSpecies = 0; iSpecies < nHeavy; iSpecies++) {
+    ef                 = Enthalpy_Formation[iSpecies] - Ru/MolarMass[iSpecies]*Ref_Temperature[iSpecies];
+    val_dPdU[iSpecies] = T*Ru/MolarMass[iSpecies] + Ru*conc/rhoCvtr *
+                         (-Cvtrs[iSpecies]*(T-Ref_Temperature[iSpecies]) -
+                         ef + 0.5*sqvel);
+
+  }
+  if (ionization) {
+    for (iSpecies = 0; iSpecies < nHeavy; iSpecies++) {
+      //      evibs = Ru/MolarMass[iSpecies] * thetav[iSpecies]/(exp(thetav[iSpecies]/Tve)-1.0);
+      //      num = 0.0;
+      //      denom = g[iSpecies][0] * exp(-thetae[iSpecies][0]/Tve);
+      //      for (iEl = 1; iEl < nElStates[iSpecies]; iEl++) {
+      //        num   += g[iSpecies][iEl] * thetae[iSpecies][iEl] * exp(-thetae[iSpecies][iEl]/Tve);
+      //        denom += g[iSpecies][iEl] * exp(-thetae[iSpecies][iEl]/Tve);
+      //      }
+      //      eels = Ru/MolarMass[iSpecies] * (num/denom);
+
+      val_dPdU[iSpecies] -= rho_el * Ru/MolarMass[nSpecies-1] * (val_eves[iSpecies])/rhoCvve;
+    }
+    ef = Enthalpy_Formation[nSpecies-1] - Ru/MolarMass[nSpecies-1]*Ref_Temperature[nSpecies-1];
+    val_dPdU[nSpecies-1] = Ru*conc/rhoCvtr * (-ef + 0.5*sqvel)
+        + Ru/MolarMass[nSpecies-1]*Tve
+        - rho_el*Ru/MolarMass[nSpecies-1] * (-3.0/2.0*Ru/MolarMass[nSpecies-1]*Tve)/rhoCvve;
+  }
+
+  /*--- Momentum derivatives ---*/
+  for (iDim = 0; iDim < nDim; iDim++)
+    val_dPdU[nSpecies+iDim] = -conc*Ru*V[VEL_INDEX+iDim]/rhoCvtr;
+
+  /*--- Total energy derivative ---*/
+  val_dPdU[nSpecies+nDim]   = conc*Ru / rhoCvtr;
+
+  /*--- Vib.-el energy derivative ---*/
+  val_dPdU[nSpecies+nDim+1] = -val_dPdU[nSpecies+nDim] +
+                               rho_el*Ru/MolarMass[nSpecies-1]*1.0/rhoCvve;    
+}
+
+void CNEMOGas::GetdTdU(su2double *V, su2double *val_dTdU){
+
+  su2double v2, ef, rhoCvtr;
+
+  /*--- Necessary indexes to assess primitive variables ---*/
+  unsigned long VEL_INDEX     = nSpecies+2;
+  unsigned long RHOCVTR_INDEX = nSpecies+nDim+6;
+
+  /*--- Rename for convenience ---*/
+  rhoCvtr = V[RHOCVTR_INDEX];
+
+  Cvtrs              = GetSpeciesCvTraRot();
+  Enthalpy_Formation = GetSpeciesFormationEnthalpy();
+  Ref_Temperature    = GetRefTemperature();
+
+  /*--- Calculate supporting quantities ---*/
+  v2 = 0.0;
+  for (iDim = 0; iDim < nDim; iDim++)
+    v2 += V[VEL_INDEX+iDim]*V[VEL_INDEX+iDim];
```

Comment:
Does this require a CGeometry class to be passed into function?

Reply:
This may be a dumb question....
would it be okay to include the geometrytool box in the fluid model to do this for a single computation? 

The NEMO team needs to spend some time making use of the toolbox.

Reply:
I think so, it is a very small header with just some functions, the benefit is knowing immediately what is being computed from just the function name, not comments needed or anything.
You can do "using namespace GeometryToolbox;" to simplify things a bit, but do that only in cpp files (if you do it in hpp it spreads and eventually may cause ambiguous function names.